pkg/kubelet/container/runtime.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/volume"
+)
+
+// Runtime interface defines the interfaces that should be implemented
+// by a container runtime.
+type Runtime interface {
+	// Version returns a map of version information of the container runtime.
+	Version() (map[string]string, error)
+	// ListPods returns a list of Pod. The boolean parameter specifies whether
+	// the runtime returns all pods including those already exited and dead pods(used
+	// for garbage collection).
+	ListPods(all bool) ([]*api.Pod, error)
```

Comment:
@vmarmol Good poing, AFAIK, the log is still available via journal or something. We only need to filter the individual container's. Let me ping rocket team to confirm that :)

@dchen1107 I mean by adding `ID` field in `api.Container`, then when we do `ListPods(true)` in GC, we can have the those dead containers also appear in the same pod with running ones. Just like old days, the total number of containers are the same, the only difference is they are grouped by the pod's.